corona-warn-app/cwa-app-android
# Refactoring & state-aware illustrations for Settings>BackgroundPriority (EXPOSUREAPP-3632)

Settings>BackgroundPriority  illustrations were not state aware, they always showed the same icon + color independent of the current background priority settings.

This PR adds new illustrations with matching icons (enabled/green + disabled/red).

When I wanted write logic I was looking for some kind of `backgroundPriority: Flow<Boolean>` and noticed the :spaghetti: around `ConnectivityHelper` again, which was still up for refactoring. 

I moved relevant logic into `BackgroundModeStatus`, and deleted the now superfluos code from ConnectivityHelper, see:
https://github.com/corona-warn-app/cwa-app-android/pull/1997/files#diff-7e2414bf7818e57619fd3c9606df0f476447adaac4f9fc775bf18b699a38c9a7


## Testing
* Check the dialogs that pop up on first install
* Check state changes in settings
